File: 프로그래머스/1/12901. 2016년/2016년.java
import java.util.*;

class Solution {
    public String solution(int a, int b) {
        Map<Integer, String> map = new HashMap<>();
        map.put(0, "FRI");
        map.put(1, "SAT");
        map.put(2, "SUN");
        map.put(3, "MON");
        map.put(4, "TUE");
        map.put(5, "WED");
        map.put(6, "THU");

        int[] days = new int[] { 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
        int day = 0;
        for (int i = 0; i < a - 1; i++) {
            day += days[i];
        }
        day += b - 1;

        return map.get(day % 7);
    }
}

File: 프로그래머스/1/12901. 2016년/2016년.py
def solution(a, b):
    answer = ''
    days = [31,29,31,30,31,30,31,31,30,31,30,31]
    day = dict(enumerate(days, start=1))
    count = 0
    month = 1
    while month < a:
        count += days[month-1]
        month += 1
    count += b
    name = ['FRI','SAT','SUN','MON','TUE','WED','THU']
    return name[(count-1) % len(name)]
